### PR TITLE
IGNITE-16258 Support Serializable lambdas marshalling in User Object Serialization

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-16258

Switched from `MethodHandle` to `Method`, because in the non-static case `Method` performs better than `MethodHandle`

```
Benchmark                                                                       Mode  Cnt   Score   Error  Units
MethodVsMethodHandleBenchmark.directInexactMethodHandleReturningNewObject       avgt   15   8,130 ± 0,371  ns/op
MethodVsMethodHandleBenchmark.directInexactMethodHandleReturningSelf            avgt   15   6,322 ± 0,076  ns/op
MethodVsMethodHandleBenchmark.directExactMethodHandleReturningNewObject         avgt   15  11,203 ± 3,175  ns/op
MethodVsMethodHandleBenchmark.directExactMethodHandleReturningSelf              avgt   15  10,601 ± 0,221  ns/op
MethodVsMethodHandleBenchmark.empty                                             avgt   15   4,441 ± 0,058  ns/op
MethodVsMethodHandleBenchmark.methodReturningNewObject                          avgt   15  14,832 ± 0,377  ns/op
MethodVsMethodHandleBenchmark.methodReturningSelf                               avgt   15   5,359 ± 0,074  ns/op
MethodVsMethodHandleBenchmark.unreflectedInexactMethodHandleReturningNewObject  avgt   15   8,691 ± 1,347  ns/op
MethodVsMethodHandleBenchmark.unreflectedInexactMethodHandleReturningSelf       avgt   15   8,661 ± 2,511  ns/op
MethodVsMethodHandleBenchmark.unreflectedExactMethodHandleReturningNewObject    avgt   15  13,074 ± 0,172  ns/op
MethodVsMethodHandleBenchmark.unreflectedExactMethodHandleReturningSelf         avgt   15  10,476 ± 0,162  ns/op
```

Also, `MethodHandle` does not work for calling `SerializedLambda#readResolve()`.

### CODE REVIEW

File: modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/Classes.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+/**
+ * Utilities to work with classes.
+ */
+class Classes {
+    static boolean isSerializable(Class<?> objectClass) {
+        return Serializable.class.isAssignableFrom(objectClass);
+    }
+
+    static boolean isLambda(Class<?> objectClass) {
+        return objectClass.getSimpleName().contains("$$Lambda$") && objectClass.isSynthetic()
```
Comment (SammyVimes) [inline modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/Classes.java:31]: I wonder if this "$$Lambda$" is JVM specific

**Comment by semyon** [comment]
Gentlemen, I think you've tagged a wrong Semyon :)

**Comment by rpuch** [comment]
@semyon Indeed, sorry for that :)

File: modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/Classes.java (hunk 1 above)
Comment (rpuch) [reply modules/network/src/main/java/org/apache/ignite/internal/network/serialization/marshal/Classes.java]: Yes, it's JVM specific. Actually, when checking for a lambda here, we actually want to know that the class cannot be loaded by name, so it makes sense to check this property directly.

Reworked the fix to detect whether we can load the class by name (and that it's not a primitive or an array); along with `isSynthetic()` it should yield the correct result.

**Review by SammyVimes — APPROVED** [review]
LGTM

**Pull request merged**